client/src/components/App/App.tsx

```diff
@@ -1,32 +1,72 @@
 import "./App.less";
 
-import React from "react";
-import { Route, Switch } from "react-router-dom";
+import axios from "axios";
+import React, { useEffect, useState } from "react";
+import { Redirect, Route, Switch } from "react-router-dom";
 
+import { isLoggedInPath, IsLoggedInResponse } from "../../api";
 import DashboardPage from "../Dashboard/DashboardPage";
 import FamiliesPage from "../Families/FamiliesPages";
 import Header from "../Header/Header";
+import LoginPage from "../Login/LoginPage";
 import MatesPage from "../Mates/MatesPage";
 import NewMatesPage from "../NewMates/NewMatesPage";
 
-const App = (): JSX.Element => (
-  <>
-    <Header />
-    <Switch>
-      <Route path="/mates">
-        <MatesPage />
-      </Route>
-      <Route path="/families">
-        <FamiliesPage />
-      </Route>
-      <Route path="/add-mates">
-        <NewMatesPage />
-      </Route>
-      <Route path="/">
-        <DashboardPage />
-      </Route>
-    </Switch>
-  </>
+type ProtectedRoutesPropTypes = {
+  children: React.ReactNode;
+  loggedIn: boolean;
+  path: string;
+};
+
+const ProtectedRoute = ({
+  children,
+  loggedIn,
+  path,
+}: ProtectedRoutesPropTypes) => (
+  <Route path={path}>{loggedIn ? children : <Redirect to="/login" />}</Route>
 );
 
+const App = (): JSX.Element => {
+  const [loggedIn, setLoggedIn] = useState(false);
+
+  useEffect(() => {
+    axios
+      .get<IsLoggedInResponse>(isLoggedInPath)
+      .then((res) => {
+        let resLoggedIn = false;
+        if ("isLoggedIn" in res.data) {
+          resLoggedIn = res.data.isLoggedIn;
+        }
+        if (loggedIn !== resLoggedIn) {
+          setLoggedIn(resLoggedIn);
+        }
+      })
+      .catch((err) => console.error(err));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
It can, but you would have to pass in both the state to act upon and the function to call to update the state. 

IMO the logic on how to update the state is tightly coupled with the specific component so it wouldn't make sense to write that code in a helper class.